required/latex-lab/latex-lab-block.dtx

```diff
@@ -2210,12 +2271,26 @@
       \tl_set_eq:NN \@itemlabel \l_@@_item_label_tl
     }
 %    \end{macrocode}
-%    finally, we signal that we are at the start of a new list (which
+%    Finally, we signal that we are at the start of a new list (which
 %    effects how the first \cs{item} is handled and how \cs{par}
 %    commands are interpreted.
 %    \begin{macrocode}
     \legacy_if_gset_true:n { @newlist }
 %    \end{macrocode}
+%    If we encounter horizontal material before the first \cs{item} we
+%    do want a \cs{@noitemerr} straight away, because afterwards we
+%    end up with tagging structure faults  and the cause is the
+%    missing \cs{item}. So we setup up \cs{@@_item_everypar:} to test
+%    for this. When the first \cs{item} is encountered this is then
+%    reset. This is only relevant for vertical lists, if we have
+%    inline lists one would need to test for somethingg else to
+%    identify that there is horizontal material between list start and
+%    the first \cs{item}, maybe some \cs{spacefactor} trick could be
+%    used then.\fmi{Think about a better implementation at some point.}
```

Comment:
As we know the first thing after the start of the env. must be `\item`, I suspect a look-ahead might be easier ...

Reply:
that is not true, e.g. this is quite allowed:
~~~~
\begin{enumerate} 
\setcounter{enumi}{100}
\item blub
\end{enumerate}
~~~~

Reply:
Well that should clearly be `\begin{enumerate}[start = 100]` or similar :)

More seriously, one could check that the result of boxing material between the start of the env and the first `\item` has no width.

Reply:
I doubt it. the item might be hidden as in `\bibitem` or worse. On the other hand, it might be the only thing reliable. The spacefact trick should work most of the time except for characters with spacefactor 0. Might be acceptable, might be not ... we'll see

Reply:
I'm reasonably serious about looking ahead - it would be easy enough to cover `\bibitem` as well as `\item`, and one could have redirection for older documents but insist that lists have to start `\(bib)item` with `\DocumentMetadata`

Reply:
Anyway, that's an aside - at present the PR doesn't change the behaviour here

Reply:
> ;-). But then `\makebox[0pt]{blub}` would be allowed ... or a full overlay tikzpicture.

not true. any box changes spacefactor to 1000 ... see TeX book. So if you start with spacesfactor that is out of range you see any hmode stuff unless it is ")" and one or two others

Reply:
> Well that should clearly be `\begin{enumerate}[start = 100]` or similar :)
> 
> More seriously, one could check that the result of boxing material between the start of the env and the first `\item` has no width.

Testing is neat idea, except for a bad optimisation of TeX: anything that goes in an \hbox gets no discretionaries added and unbox it later doesn't help unfortunately if I remember correctly.

Reply:
Nope, `\discretionary` works fine:

```latex
\documentclass{article}

\begin{document}
\hsize=2cm
\sbox0{This is important crabbable.}\unhbox0

\sbox0{This is im\discretionary{i}{}{}portant crab\discretionary{i}{}{}bable.}\unhbox0
\end{document}
```

Reply:
nope  partly .... :-) I didn't mean explicit hyphens, but I remembered incorrectly. it is not the hyphen that has a problem (as they are added only when the paragraph is finally built) but the language nodes that are omitted in an hbox.

Try this:
```
\documentclass{article}

\begin{document}

\def\sample{%
A few words in one language \language1
A few words in one language \language2
A few words in one language \language0}

\sample

\setbox0\hbox{\sample}

\unhbox0

\showoutput

\end{document}
```

Reply:
"the language nodes that are omitted in an hbox."
How weird! "No bad language please, as we are boxed in!"

Reply:
One of the few (I would say) misguided optimizations, I guess the idea was that in hboxes there will be no hyphenation thus avoid extra processing.